src/main/java/me/ryanhamshire/GPFlags/util/ChatUtil.java

```diff
@@ -0,0 +1,258 @@
+        Pattern pattern = Pattern.compile("&[a-flmnokrA-F0-9]");
+        Matcher matcher = pattern.matcher(message);
+        while (matcher.find()) {
+            String colorCode = message.substring(matcher.start(), matcher.end());
+            String replaceSharp = colorCode.replace("&", "");
+
+            message = message.replace(colorCode, getNativeColor(replaceSharp.toCharArray()[0]));
+            matcher = pattern.matcher(message);
+        }
+        return message;
+    }
+
+    public static Component fixColor(String message, String key, String value) {
+        if (message.split(key).length == 0)
+            return ChatUtil.hexComp(value);
+
+        String colorMsg = message.split(key)[0];
+        String color = colorMsg.substring(colorMsg.length() - 2);
+        if (color.trim().isEmpty())
+            return ChatUtil.hexComp(value);
+        if (color.charAt(0) == '&')
+            return ChatUtil.hexComp(color + value);
+        else return ChatUtil.hexComp(value);
+    }
+
+    private static String getNativeColor(char color) {
+        String reset = "<!b><!i><!obf><!u><!st>";
+        String colorCode = "<color:white>";
+        if (color == 'r') return reset + colorCode;
+        switch (color) {
+            case 'a':
+                colorCode = "<color:green>";
+            case 'b':
+                colorCode = "<color:aqua>";
+            case 'c':
+                colorCode = "<color:red>";
+            case 'd':
+                colorCode = "<color:light_purple>";
+            case 'e':
+                colorCode = "<color:yellow>";
+            case 'f':
+                colorCode = "<color:white>";
+            case '0':
+                colorCode = "<color:black>";
+            case '1':
+                colorCode = "<color:dark_blue>";
+            case '2':
+                colorCode = "<color:dark_green>";
+            case '3':
+                colorCode = "<color:dark_aqua>";
+            case '4':
+                colorCode = "<color:dark_red>";
+            case '5':
+                colorCode = "<color:dark_purple>";
+            case '6':
+                colorCode = "<color:gold>";
+            case '7':
+                colorCode = "<color:gray>";
+            case '8':
+                colorCode = "<color:dark_gray>";
+            case '9':
+                colorCode = "<color:blue>";
+            case 'l':
+                colorCode = "<bold>";
+            case 'm':
+                colorCode = "<strikethrough>";
+            case 'n':
+                colorCode = "<u>";
+            case 'o':
+                colorCode = "<italic>";
+            case 'k':
+                colorCode = "<obf>";
+        }
+        switch (color) {
+            case 'l':
+            case 'm':
+            case 'n':
+            case 'o':
+            case 'k':
+                break;
+            default:
+                colorCode = reset + colorCode;
```

Comment:
If someone does &l&n, the following text should be both bolded and underlined, but it looks like you throw in the reset character when its not needed, making it so it would only be underlined. 

I think you're overcomplicating this method.

// The colors
if (color='0') return "\<black>";
if (color='1') return "\<dark_blue>";
if (color='2') return "\<dark_green>";
...
// And the formatting
if (color='r') return "\<reset>";
if (color='l') return "\<bold>";
if (color='m') return "\<strikethrough>";
...
// If invalid color, just return it as is
return color;

You could make a hashmap that stores the translated values but I don't think its necessary

Reply:
This i not correct. Because you split the String under `ChatUtil.parseNativeColorCodes(String message)`. He takes the pattern of the color-code, replaced each & and gets the character from the code when existent. This way, when `ChatUtil.getNativeColor` has a valid entry, he does the formatting. 

The default there seems kind of confusing and in the end it looks we could try to remove it. My case required it but I dont know why anymore. Maybe i'll work out without.